scaleway/resource_lb_certificate_beta.go

```diff
@@ -0,0 +1,204 @@
+package scaleway
+
+import (
+	"errors"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/lb/v1"
+)
+
+func resourceScalewayLbCertificateBeta() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceScalewayLbCertificateBetaCreate,
+		Read:          resourceScalewayLbCertificateBetaRead,
+		Update:        resourceScalewayLbCertificateBetaUpdate,
+		Delete:        resourceScalewayLbCertificateBetaDelete,
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"lb_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The load-balancer ID",
+			},
+			"name": {
+				Default:     newRandomName("lb-cf"),
+				Type:        schema.TypeString,
+				Description: "The name of the load-balancer certificate",
+				Optional:    true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
Correct me if I am wrong: in case we set name of certificate to be empty it would regenerate  the name (and make a patch request) on every `terraform apply`. My thinking here was that in case we do not specify a name (or we had it at some point and then remove it), then ignore that field entirely and leave it as it is. 

I might be wrong about the behaviour of terraform with default: random behaviour though

Reply:
Ho sorry, you can remove the `Default` and the `DiffSuppressFunc` for the `name` field of the schema.

It user don't fille the name, [this line](https://github.com/terraform-providers/terraform-provider-scaleway/pull/396/files#diff-c73383c35592f6d39875641bf4429df7R123) will generate it (with the `lb-cert` prefix).

If you want an example, in `instance_server`:
- [here is](https://github.com/terraform-providers/terraform-provider-scaleway/blob/master/scaleway/resource_instance_server.go#L28-L33) the `name` attribute definition
- [here is](https://github.com/terraform-providers/terraform-provider-scaleway/blob/master/scaleway/resource_instance_server.go#L250) the expansion or generation of the server name.

Reply:
Fair enough, missed that point.